packages/core/src/telemetry/telemetry.ts

```diff
@@ -113,10 +113,14 @@ export class Telemetry {
     });
   }
 
-  static setBaggage(baggage: Record<string, string>, ctx: Context = otlpContext.active()) {
-    const currentBaggage = propagation.getBaggage(ctx);
-    // @ts-ignore
-    const newCtx = propagation.setBaggage(ctx, { ...currentBaggage, ...baggage });
+  static setBaggage(baggage: Record<string, BaggageEntry>, ctx: Context = otlpContext.active()) {
+    const newCtx = propagation.setBaggage(
+      ctx,
+      propagation.createBaggage({
+        ...Object.fromEntries(propagation.getBaggage(ctx)?.getAllEntries() ?? []),
+        ...baggage,
+      }),
+    );
```

Comment:
improves readability a little
```suggestion
    const currentBaggage = Object.fromEntries(propagation.getBaggage(ctx)?.getAllEntries() ?? [])
    const newCtx = propagation.setBaggage(
      ctx,
      propagation.createBaggage({
        ...currentBaggage,
        ...baggage,
      }),
    );
```